### ISSUE TITLE
Could not pass event PlayerJoinEvent (CMI Vanish)

### ISSUE THREAD

**Issue by Dominiiikk** [Insprill/custom-join-messages#55]
### Server Version

Universespigot 1.20.4 git ex3febd

### CJM Version

17.4.0

### Describe the Bug

Well, I get this error on join and on leave with CJM and CMI. I dont event use CMI's vanish feature and I have it disabled in modules.yml inside CMI. I use PremiumVanish.

I also disabled fake messages in CJM config.

```
23:49:57] [Server thread/ERROR]: Could not pass event PlayerJoinEvent to CustomJoinMessages v17.4.0
--
2234 | java.lang.NoClassDefFoundError: com/Zrips/CMI/commands/list/vanishedit$VanishAction
2235 | at net.insprill.cjm.compatibility.cmi.CmiVanishHook.isVanished(CmiVanishHook.kt:25) ~[custom-join-messages-17.4.0.jar:?]
2236 | at net.insprill.cjm.compatibility.hook.HookManager.isVanished(HookManager.kt:16) ~[custom-join-messages-17.4.0.jar:?]
2237 | at net.insprill.cjm.message.MessageSender.trySendMessages(MessageSender.kt:51) ~[custom-join-messages-17.4.0.jar:?]
2238 | at net.insprill.cjm.listener.JoinEvent.onPlayerJoin(JoinEvent.kt:21) ~[custom-join-messages-17.4.0.jar:?]
2239 | at com.destroystokyo.paper.event.executor.asm.generated.GeneratedEventExecutor1308.execute(Unknown Source) ~[?:?]
2240 | at org.bukkit.plugin.EventExecutor$2.execute(EventExecutor.java:77) ~[?:git-UniverseSpigot-"ec3febd"]
2241 | at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:76) ~[?:git-UniverseSpigot-"ec3febd"]
2242 | at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[?:git-UniverseSpigot-"ec3febd"]
2243 | at io.papermc.paper.plugin.manager.PaperEventManager.callEvent(PaperEventManager.java:58) ~[?:git-UniverseSpigot-"ec3febd"]
2244 | at io.papermc.paper.plugin.manager.PaperPluginManagerImpl.callEvent(PaperPluginManagerImpl.java:126) ~[?:git-UniverseSpigot-"ec3febd"]
2245 | at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:617) ~[?:git-UniverseSpigot-"ec3febd"]
2246 | at net.minecraft.server.players.PlayerList.placeNewPlayer(PlayerList.java:353) ~[?:git-UniverseSpigot-"ec3febd"]
2247 | at net.minecraft.server.network.ServerConfigurationPacketListenerImpl.handleConfigurationFinished(ServerConfigurationPacketListenerImpl.java:134) ~[?:git-UniverseSpigot-"ec3febd"]
2248 | at net.minecraft.network.protocol.configuration.ServerboundFinishConfigurationPacket.handle(ServerboundFinishConfigurationPacket.java:18) ~[?:git-UniverseSpigot-"ec3febd"]
2249 | at net.minecraft.network.protocol.configuration.ServerboundFinishConfigurationPacket.a(ServerboundFinishConfigurationPacket.java:9) ~[?:git-UniverseSpigot-"ec3febd"]
2250 | at net.minecraft.network.protocol.PacketUtils.lambda$ensureRunningOnSameThread$0(PacketUtils.java:53) ~[?:?]
2251 | at net.minecraft.server.TickTask.run(TickTask.java:18) ~[?:git-UniverseSpigot-"ec3febd"]
2252 | at net.minecraft.util.thread.BlockableEventLoop.doRunTask(BlockableEventLoop.java:152) ~[?:?]
2253 | at net.minecraft.util.thread.ReentrantBlockableEventLoop.doRunTask(ReentrantBlockableEventLoop.java:24) ~[?:?]
2254 | at net.minecraft.server.MinecraftServer.doRunTask(MinecraftServer.java:1475) ~[?:git-UniverseSpigot-"ec3febd"]
2255 | at net.minecraft.server.MinecraftServer.d(MinecraftServer.java:197) ~[?:git-UniverseSpigot-"ec3febd"]
2256 | at net.minecraft.util.thread.BlockableEventLoop.pollTask(BlockableEventLoop.java:125) ~[?:?]
2257 | at net.minecraft.server.MinecraftServer.pollTaskInternal(MinecraftServer.java:1452) ~[?:git-UniverseSpigot-"ec3febd"]
2258 | at net.minecraft.server.MinecraftServer.pollTask(MinecraftServer.java:1380) ~[?:git-UniverseSpigot-"ec3febd"]
2259 | at net.minecraft.util.thread.BlockableEventLoop.managedBlock(BlockableEventLoop.java:135) ~[?:?]
2260 | at net.minecraft.server.MinecraftServer.waitUntilNextTick(MinecraftServer.java:1358) ~[?:git-UniverseSpigot-"ec3febd"]
2261 | at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1247) ~[?:git-UniverseSpigot-"ec3febd"]
2262 | at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:324) ~[?:git-UniverseSpigot-"ec3febd"]
2263 | at java.lang.Thread.run(Unknown Source) ~[?:?]
2264 | Caused by: java.lang.ClassNotFoundException: com.Zrips.CMI.commands.list.vanishedit$VanishAction
2265 | ... 29 more
```

### Expected Behavior

Expected no error and sending the custom join message.

### Steps to Reproduce

Install CMI, CJM on 1.20.4 server
Disable Vanish feature in modules.yml inside CMI
Restart the server
Join the server

### Server Logs

https://mclo.gs/1IIuzxC

### Checklist

- [X] I am running the latest version of the plugin
- [X] I have read the wiki to ensure it's not a configuration issue
- [ ] I ticked all of the boxes without actually reading them

**Comment by Insprill** [comment]
Does this happen if you enable vanish in CMI? Most likely, it's just an update that broke compatibility.

**Comment by Dominiiikk** [comment]
It does happen event with Vanish enabled. I'm running CMI 9.7.0.2

**Comment by Insprill** [comment]
I just pushed a fix. Can you download it from the action below and test it out? If all is well then I'll release the update tomorrow.
https://github.com/Insprill/custom-join-messages/actions/runs/8517953590/artifacts/1376452376

**Comment by Dominiiikk** [comment]
> I just pushed a fix. Can you download it from the action below and test it out? If all is well then I'll release the update tomorrow. https://github.com/Insprill/custom-join-messages/actions/runs/8517953590/artifacts/1376452376

Hi, I can confirm, that this seems to have fixed the issue. :)

**Comment by Dominiiikk** [comment]
Sorry for hiding that comment. I've tested it out on my acutall server and its really working. Thanks 👍

**Comment by Insprill** [comment]
Awesome, glad to hear. I'll get the update out tonight.

**Comment by Insprill** [comment]
Fixed in 17.4.1 released last night.
